### PR TITLE
convert-networking-config for Ingress

### PR DESCRIPTION
**Pull request by esnible**
Restored 'convert-networking-config' function, but just for Ingress.

### CODE REVIEW

**Comment by esnible** [comment]
@costinm When restoring this code I duplicated ConvertIngressV1alpha3() because https://github.com/istio/istio/commit/cd2c58a92a95f43d2f88386f12e6cf3b59624059#diff-d27df99ee76e76a363d5da2a3babe772 removes the generation of VirtualService from Ingress.  Do you agree that VirtualService should be generated by the CLI conversion tool?

**Comment by costinm** [comment]
Failing tests.

File: istioctl/cmd/istioctl/convert_ingress.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	inFilenames        []string
+	outConvertFilename string
+	convertIngressCmd  = &cobra.Command{
+		Use:   "convert-networking-config",
+		Short: "Convert networking configs from v1alpha1 to v1alpha3",
+		Long: "Converts sets of v1alpha1 configs to v1alpha3 equivalents on a best effort basis. " +
```
Comment (ayj) [inline istioctl/cmd/istioctl/convert_ingress.go]: Can you update the docs, examples, etc. here to make it clear this only handles ingress-to-<gateway/virtualservice> conversion. The previous v1alpha1 support has been removed.

File: istioctl/cmd/istioctl/convert_ingress.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	inFilenames        []string
+	outConvertFilename string
+	convertIngressCmd  = &cobra.Command{
+		Use:   "convert-networking-config",
```
Comment (rshriram) [inline istioctl/cmd/istioctl/convert_ingress.go]: rename to convert-ingress-to-gateway

File: istioctl/cmd/istioctl/convert_ingress.go (hunk 1 above)
Comment (rshriram) [reply istioctl/cmd/istioctl/convert_ingress.go]: +1. Also state taht this is a best effort tool and that the translations need to be inspected manually before applying

File: istioctl/cmd/istioctl/convert_ingress.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+			return convertConfigs(readers, writer)
+		},
+	}
+)
+
+func convertConfigs(readers []io.Reader, writer io.Writer) error {
+	configDescriptor := model.ConfigDescriptor{
+		model.VirtualService,
+		model.Gateway,
+		model.ServiceEntry,
```
Comment (rshriram) [inline istioctl/cmd/istioctl/convert_ingress.go]: you only need the first two

File: istioctl/cmd/istioctl/testdata/v1alpha1/another-ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: extensions/v1beta1
```
Comment (rshriram) [inline istioctl/cmd/istioctl/testdata/v1alpha1/another-ingress.yaml]: testdata/ingress/..

File: istioctl/pkg/convert/ingress.go [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+	out = append(out, virtualServices...)
+
+	return out, nil
+}
+
+// convertIngressV1alpha3 (and the two functions below that it calls) could be replaced with
+// the version in _pilot/pkg/kube/ingress/conversion.go_ if the generation of VirtualServices
+// was restored.
+func convertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model.Config, model.Config) {
+	gateway := &networking.Gateway{
```
Comment (rshriram) [inline istioctl/pkg/convert/ingress.go]: ???
We already have all this in k8s ingress code. Just reuse those functions.

Comment (esnible) [reply istioctl/pkg/convert/ingress.go]: We did.  https://github.com/istio/istio/commit/cd2c58a92a95f43d2f88386f12e6cf3b59624059#diff-d27df99ee76e76a363d5da2a3babe772 removed the generation of VirtualService from that method.  I could not find a replacement...

**Comment by linsun** [comment]
/lgtm
thanks @costinm for triggering another test.  mixer 2e2 failing should not be caused by istioctl changes.

**Comment by esnible** [comment]
@costinm I am now only generating VirtualService.  I am using the pilot function.  I am post-processing the VirtualService name if it begins with "-" because the pilot function generates the name "-simple-ingress-istio-autogenerated-k8s-ingress" which is not valid Kubernetes.  Possibly that change should go into the function instead of being applied in the CLI code.
@rshriram Because I only generate VirtualService, I changed the name from 'convert-ingress-to-gateway' to 'convert-ingress'.

**Pull request merged**